### PR TITLE
NCSDK-26629: Add suit subsystem

### PR DESCRIPTION
**Pull request by tomchy**
Add SUIT subsystem copy (as-is) without alignment to the NCS.

### CODE REVIEW

**Review by ahasztag — APPROVED** [review]

**Review by adsz-nordic — COMMENTED** [review]
Went through the files briefly, as many of them are copied/moved.
Spotted a potential license mismatch reported by github - please double check it.

**Review by robertstypa — APPROVED** [review]

File: tests/subsys/suit/common/tls_config/user-tls-conf.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/*
```
[the remark this answers is not in the document]
Comment (tomchy) [reply tests/subsys/suit/common/tls_config/user-tls-conf.h:1]: Yes, it is correct and accepted to merge despite the red status of the license check.

**Review by adsz-nordic — APPROVED** [review]

**Review by carlescufi — DISMISSED** [review]
There is a fixup commit in there. Please just merge it with the corresponding one, since this is introducing a new subsystem

**Comment by carlescufi** [comment]
Also is there a particular reason why you are introducing new error types instead of using standard `-Exx` ?

**Comment by tomchy** [comment]
> There is a fixup commit in there. Please just merge it with the corresponding one, since this is introducing a new subsystem

Commits squashed, as requested

**Comment by tomchy** [comment]
> Also is there a particular reason why you are introducing new error types instead of using standard `-Exx` ?

There are a couple of reasons:
 - SUIT processor introduces it's own error types as it is not built to be dependent on Zephyr
 - SUIT platform handles and returns SUIT processor codes, so it is cleaner to have a proper mapping between platform and processor codes.
 - Whenever possible, we use the same values as Zephyr errors, to make things easier.
 - We wanted to have a well defined return type for each platform function, so it will be possible to manage error cases efficiently. One benefit is that it is now clear if you return platform code, processor code or other subsys (Zephyr) code and when to do the translation.

The error codes were introduced and are deeply incorporated inside the code. If you think that the justification is not sufficient, I propose to schedule a meeting, discuss it with the whole team and plan such changes accordingly, but do not block the PR. It will significantly delay the whole migration process otherwise.

**Comment by carlescufi** [comment]
License check failure is just Apache v2.

**Pull request merged**